### PR TITLE
Feature/advanced logging

### PR DESCRIPTION
**Pull request by Bukowskaii**
advanced logging for map access, including ip addresses and geolocation courtesy of ip-api.com

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,11 +34,13 @@
     "express-rate-limit": "^5.1.3",
     "express-session": "^1.17.1",
     "extend": "^3.0.2",
+    "geoip-lite": "^1.4.0",
```
Comment (Bukowskaii) [inline package.json:37]: not necessary, originally was using geoip-lite but switched to web based API from ip-api.com

File: src/routes/discord.js [hunk 2]
```diff
@@ -3,7 +3,8 @@
 const express = require('express');
 const axios = require('axios');
 const router = express.Router();
-
+const fetch = require('node-fetch');
+const geoip = require('geoip-lite');
```
Comment (Bukowskaii) [inline src/routes/discord.js:7]: see package.json line 37 comment

Comment (versx) [inline src/routes/discord.js:7]:
```suggestion
```

File: package.json (hunk 1 above)
Comment (versx) [inline package.json:37]:
```suggestion
```

**Review by versx — APPROVED** [review]
Looks good

**Comment by Bukowskaii** [comment]
nice changes

**Comment by versx** [comment]
> nice changes

Sorry had to clean it up just a little, love this PR though. The more details the better imo, running this PR myself and really like the spoiler for the IP address. (as well as all the info)

**Comment by Bukowskaii** [comment]
> > nice changes
> 
> Sorry had to clean it up just a little, love this PR though. The more details the better imo, running this PR myself and really like the spoiler for the IP address. (as well as all the info)

no apologies, it's much cleaner and agile now!

**Comment by Bukowskaii** [comment]
For us the client info is the biggest thing, seeing what device(s) our users are using. I'm not super satisfied with the geoip, but I'm going to be looking at doing an ajax callback from the map frontend to get actual device location on login

**Review by TurtIeSocks — APPROVED** [review]

**Pull request merged**